runtime/src/in_mem_accounts_index.rs

```diff
@@ -806,31 +806,44 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
         thread_rng().gen_range(0, N) == 0
     }
 
+    /// assumes 1 entry in the slot list. Ignores overhead of the HashMap and such
+    fn approx_size_of_one_entry() -> usize {
+        std::mem::size_of::<T>()
+            + std::mem::size_of::<Pubkey>()
+            + std::mem::size_of::<AccountMapEntry<T>>()
+    }
+
     /// return true if 'entry' should be removed from the in-mem index
     fn should_remove_from_mem(
         &self,
         current_age: Age,
         entry: &AccountMapEntry<T>,
         startup: bool,
         update_stats: bool,
+        exceeds_budget: bool,
```

Comment:
To me it feels like the `exceeds_budget` should not be part of this function, and instead is used alongside instead.

I.e. currently
```rust
if self.should_remove_from_mem(current_age, v, startup, true, exceeds_budget) {
                        removes.push(*k);
}
```

My suggestion:
```rust
if exceeds_budget || self.should_remove_from_mem(current_age, v, startup, true) {
                        removes.push(*k);
}
```

And then the same thing in the impl of `flush_remove_from_cache()`.

Reply:
should_remove_from_mem only considers throwing things out if the item's age is ready (or at startup). So, we only use exceeds_budget after we check for age. exceeds_budget is a reason to ignore the non-age reasons we normally keep stuff in the cache.